samsa2-master.nf

```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/env nextflow
+
+params.input_dir = "$projectDir/input_files"
+params.python_scripts = "$projectDir/python_scripts"
+params.programs = "$projectDir/programs"
+params.diamond_database = "$projectDir/full_databases/New_Bac_Vir_Arc_RefSeq"
+refseq_db = "${params.diamond_database}.fa"
+params.subsys_database = "$projectDir/full_databases/subsys_db"
+params.output_dir = "$projectDir/output_files"
+step_1_output_dir = "$params.output_dir/step_1_output"
+step_2_output_dir = "$params.output_dir/step_2_output"
+step_3_output_dir = "$params.output_dir/step_3_output"
+step_4_output_dir = "$params.output_dir/step_4_output"
+step_5_output_dir = "$params.output_dir/step_5_output"
+
+input_files = "${params.input_dir}/*_R{1,2}_*" // match everything else
+
+// step 1
+process TRIMMOMATIC {
+
+    cpus 6
+    memory '12 GB'
+    publishDir step_1_output_dir, mode: 'copy'
+    module 'trimmomatic'
+
+    input:
+    tuple val(sample_id), path(reads)
+
+    output:
+    tuple val("${sample_id}"), path("${sample_id}.cleaned.forward"), path("${sample_id}.cleaned.reverse")
+
+    script:
+    """
+    trimmomatic PE -phred33 -threads $task.cpus "$reads" ${sample_id}.cleaned.forward ${sample_id}.cleaned.forward_unpaired ${sample_id}.cleaned.reverse ${sample_id}.cleaned.reverse_unpaired SLIDINGWINDOW:4:15 MINLEN:70
+    """
+
+}
+
+// step 2
+process PEAR {
+    cpus 56
+    memory '56 GB'
+    publishDir step_2_output_dir, mode: 'copy'
+    module 'pear/0.9.11'
+
+    input:
+    tuple val(sample_id), path(reads_cleaned_forward), path(reads_cleaned_reverse)
+
+    output:
+    tuple val("$sample_id"), path("${sample_id}.merged.assembled.fastq"), path("${sample_id}.merged.discarded.fastq"), path("${sample_id}.merged.unassembled.forward.fastq"), path("${sample_id}.merged.unassembled.reverse.fastq"), path("${sample_id}.merged.assembled.fastq.ribosomes.log")
+
+    script:
+    """
+    pear -f $reads_cleaned_forward -r $reads_cleaned_reverse -j $task.cpus -o ${sample_id}.merged 2>&1 | tee ${sample_id}.merged.assembled.fastq.ribosomes.log
+    """
+}
+
+// step 2.9
+process RAWREADCOUNT {
+
+    cpus 1
+    memory '4 GB'
+    publishDir step_2_output_dir, mode: 'copy'
+
+    input:
+    path(input_forward_reads)
+
+    output:
+    path 'raw_counts.txt'
+
+    script:
```

Comment:
Consider using [shell](https://www.nextflow.io/docs/latest/process.html#shell) if you are mixing bash and nf variables.

Reply:
opened issue https://github.com/WEHI-ResearchComputing/samsa2-nf/issues/4